diff -ruN mbedtls-2.2.1/library/x509.c mbedtls-2.2.1_sign/library/x509.c
--- mbedtls-2.2.1/library/x509.c	2016-07-06 14:30:43.055337116 -0400
+++ mbedtls-2.2.1_sign/library/x509.c	2016-07-06 15:53:19.582629255 -0400
@@ -561,8 +561,11 @@
 
     sig->tag = **p;
 
-    if( ( ret = mbedtls_asn1_get_bitstring_null( p, end, &len ) ) != 0 )
+    if( ( ret = mbedtls_asn1_get_bitstring_null( p, end, &len ) ) != 0 ) {
+#ifndef FUZZER_DISABLE_SIGNCHECK
         return( MBEDTLS_ERR_X509_INVALID_SIGNATURE + ret );
+#endif
+    }
 
     sig->len = len;
     sig->p = *p;
diff -ruN mbedtls-2.2.1/library/x509_crt.c mbedtls-2.2.1_sign/library/x509_crt.c
--- mbedtls-2.2.1/library/x509_crt.c	2016-07-06 14:30:43.055337116 -0400
+++ mbedtls-2.2.1_sign/library/x509_crt.c	2016-07-06 15:52:23.533130509 -0400
@@ -2226,6 +2226,7 @@
         }
     }
 
+#ifndef FUZZER_DISABLE_SIGNCHECK
     /* Check the type and size of the key */
     pk_type = mbedtls_pk_get_type( &crt->pk );
 
@@ -2234,6 +2235,7 @@
 
     if( x509_profile_check_key( profile, pk_type, &crt->pk ) != 0 )
         *flags |= MBEDTLS_X509_BADCERT_BAD_KEY;
+#endif
 
     /* Look for a parent in trusted CAs */
     for( parent = trust_ca; parent != NULL; parent = parent->next )
